graylog2-server/src/main/java/org/graylog2/entitygroups/model/DBEntityGroupService.java

```diff
@@ -0,0 +1,143 @@
+import org.graylog2.database.pagination.MongoPaginationHelper;
+import org.graylog2.database.utils.MongoUtils;
+import org.graylog2.database.utils.ScopedEntityMongoUtils;
+import org.graylog2.search.SearchQuery;
+import org.graylog2.search.SearchQueryField;
+import org.graylog2.search.SearchQueryParser;
+
+import jakarta.inject.Inject;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Filters.exists;
+import static com.mongodb.client.model.Filters.in;
+
+
+public class DBEntityGroupService {
+    public static final String COLLECTION_NAME = "entity_groups";
+
+    private static final ImmutableMap<String, SearchQueryField> ALLOWED_FIELDS = ImmutableMap.<String, SearchQueryField>builder()
+            .put(EntityGroup.FIELD_NAME, SearchQueryField.create(EntityGroup.FIELD_NAME))
+            .build();
+
+    private final MongoCollection<EntityGroup> collection;
+    private final MongoUtils<EntityGroup> mongoUtils;
+    private final ScopedEntityMongoUtils<EntityGroup> scopedEntityMongoUtils;
+    private final MongoPaginationHelper<EntityGroup> paginationHelper;
+    private final SearchQueryParser searchQueryParser;
+
+    @Inject
+    public DBEntityGroupService(MongoCollections mongoCollections,
+                                EntityScopeService entityScopeService) {
+        this.collection = mongoCollections.collection(COLLECTION_NAME, EntityGroup.class);
+        this.mongoUtils = mongoCollections.utils(collection);
+        this.scopedEntityMongoUtils = mongoCollections.scopedEntityUtils(collection, entityScopeService);
+        this.paginationHelper = mongoCollections.paginationHelper(collection);
+        this.searchQueryParser = new SearchQueryParser(EntityGroup.FIELD_ENTITIES, ALLOWED_FIELDS);
+
+        final IndexOptions caseInsensitiveOptions = new IndexOptions()
+                .collation(Collation.builder().locale("en").collationStrength(CollationStrength.SECONDARY).build())
+                .unique(true);
+        collection.createIndex(new BasicDBObject(EntityGroup.FIELD_NAME, 1), caseInsensitiveOptions);
+    }
+
+    public Optional<EntityGroup> get(String id) {
+        return mongoUtils.getById(id);
+    }
+
+    public PaginatedList<EntityGroup> findPaginated(String query, int page, int perPage, Bson sort, Predicate<EntityGroup> filter) {
+        final SearchQuery searchQuery = searchQueryParser.parse(query);
+        return filter == null ?
+                paginationHelper.filter(searchQuery.toBson()).sort(sort).perPage(perPage).page(page) :
+                paginationHelper.filter(searchQuery.toBson()).sort(sort).perPage(perPage).page(page, filter);
+    }
+
+    public EntityGroup save(EntityGroup EntityGroup) {
```

Comment:
```suggestion
    public EntityGroup save(EntityGroup entityGroup) {
```
Camel case

Reply:
Thanks for the catches! Looks like some bad find-and-replaces from the original categories verbiage.